src/infql.jl

```diff
@@ -378,3 +378,179 @@ function LazyBandedMatrices._SymTridiagonal(::Tuple{TriangularLayout{'L', 'N', P
+    end
+    F = ql(A[1:(N÷2),1:(N÷2)])
+    return (F.factors[1:50,1:50], F.τ[1:50])
+end
+
+# Resize and filling functions for cached implementation
+function resizedata!(K::AdaptiveQLFactors, nm...)
+    nm = maximum(nm)
+    νμ = K.datasize[1]
+    if nm > νμ
+        olddata = copy(K.data)
+        K.data = similar(K.data, nm, nm)
+        K.data[axes(olddata)...] = olddata
+        inds = νμ:nm
+        cache_filldata!(K, inds)
+        K.datasize = size(K.data)
+    end
+    K
+end
+
+function resizedata!(K::AdaptiveQLTau, nm...)
+    nm = maximum(nm)
+    νμ = K.datasize
+    if nm > νμ
+        resize!(K.data,nm)
+        cache_filldata!(K, νμ:nm)
+        K.datasize = size(K.data,1)
+    end
+    K
+end
+
+function cache_filldata!(A::AdaptiveQLFactors{T}, inds::UnitRange{Int}) where T
+    j = maximum(inds)
+    maxN = 1000*j # Prevent runaway loop
+    Lerr = one(T)
+    N = j
+    checkinds = max(1,j-bandwidth(A.M,1)-bandwidth(A.M,2))
+    @inbounds Ls = ql(A.M[checkinds:N,checkinds:N]).L[2:j-checkinds+1,2:j-checkinds+1]
+    @inbounds while Lerr > A.tol
+        # compute QL for small finite section and large finite section
+        Ll = ql(A.M[checkinds:2N,checkinds:2N]).L[2:j-checkinds+1,2:j-checkinds+1]
+        # compare bottom right sections and stop if desired level of convergence achieved
+        Lerr = norm(Ll-Ls,2)
+        if N == maxN
+            error("Reached max. iterations in finite section QL without convergence to desired tolerance.")
+        end
+        Ls = Ll
+        N = 2*N
+    end
+    A.data = ql(A.M[1:(N÷2),1:(N÷2)]).factors[1:j,1:j]
+end
+
+
+function cache_filldata!(A::AdaptiveQLTau{T}, inds::UnitRange{Int}) where T
+    j = maximum(inds)
+    maxN = 1000*j
+    Lerr = one(T)
+    N = j
+    checkinds = max(1,j-bandwidth(A.M,1)-bandwidth(A.M,2))
+    @inbounds Ls = ql(A.M[checkinds:N,checkinds:N]).L[2:j-checkinds+1,2:j-checkinds+1]
+    @inbounds while Lerr > A.tol
+        # compute QL for small finite section and large finite section
+        Ll = ql(A.M[checkinds:2N,checkinds:2N]).L[2:j-checkinds+1,2:j-checkinds+1]
+        # compare bottom right sections and stop if desired level of convergence achieved
+        Lerr = norm(Ll-Ls,2)
+        if N == maxN
+            error("Reached max. iterations in finite section QL without convergence to desired tolerance.")
+        end
+        Ls = Ll
+        N = 2*N
+    end
+    A.data = ql(A.M[1:(N÷2),1:(N÷2)]).τ[1:j]
+end
+
+function getproperty(F::QL{T, AdaptiveQLFactors{T}, AdaptiveQLTau{T}}, d::Symbol) where T
+    if d == :L
+        return LowerTriangular(F.factors)
+    elseif d == :Q
+        return getQ(F)
+    else
+        getfield(F, d)
+    end
+end
+
+@inline function getQ(F::QL{T, AdaptiveQLFactors{T}, AdaptiveQLTau{T}}) where T
+    Q = AdaptiveQLQ{T}(zeros(1,1),F.factors,F.τ,(0,0))
+    # initialize as 50x50, then adaptively expand
+    resizedata!(Q,50)
+    return Q
+end
+
+mutable struct AdaptiveQLQ{T} <: AbstractCachedMatrix{T}
```

Comment:
This type is completely unnecessary: just use

https://github.com/JuliaLinearAlgebra/MatrixFactorizations.jl/blob/2302d61a22e5cd50646495f7bceffd09d5596c47/src/ql.jl#L242

(which I think will be done automatically since we are using type `QL`).

Note the entries of an orthogonal matrix should never be used and therefore it is a mistake to cache them

Reply:
They are not cached unless the user explicitly asks for Q to be constructed. But I can remove it entirely if you think that feature isn't useful.

I will double check whether the QL type can construct Q from the other pieces automatically.

Reply:
That's not true, the user hasn't asked for a dense `Q` to be constructed. Just like with finite dimensional QR:
```julia
julia> n = 1000; A = randn(n,n); F = qr(A);

julia> @time F.Q # doesn't do anything since it just returns a lazy type!
  0.000006 seconds (1 allocation: 32 bytes)
1000×1000 LinearAlgebra.QRCompactWYQ{Float64, Matrix{Float64}, Matrix{Float64}}:
```
So yes remove it, the feature is inconsistent with finite dimensional QR and QL and shouldn't be used

Reply:
ok it wasn't automatic but I got it working with the MatrixFactorizations approach. now to rerun all tests and make sure it doesn't break anything else